posts/renv-complications/index.qmd

```diff
@@ -0,0 +1,131 @@
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c colOrderStats.c -o colOrderStats.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c colRanges.c -o colRanges.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c diff2.c -o diff2.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c indexByRow.c -o indexByRow.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c logSumExp.c -o logSumExp.o
+gcc -I"/usr/share/R/include" -DNDEBUG       -fpic  -g -O2 -ffile-prefix-map=/build/r-base-H0vbME/r-base-4.3.2=. -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2  -c mean2.c -o mean2.o
+In file included from mean2_lowlevel.h:14,
+                 from mean2.c:9:
+mean2_lowlevel_template.h: In function ‘mean2_int’:
+mean2_lowlevel_template.h:59:13: error: ‘DOUBLE_XMAX’ undeclared (first use in this function); did you mean ‘DBL_MAX’?
+   59 |   if (sum > DOUBLE_XMAX) {
+      |             ^~~~~~~~~~~
+      |             DBL_MAX
+mean2_lowlevel_template.h:59:13: note: each undeclared identifier is reported only once for each function it appears in
+In file included from mean2_lowlevel.h:18,
+                 from mean2.c:9:
+mean2_lowlevel_template.h: In function ‘mean2_dbl’:
+mean2_lowlevel_template.h:59:13: error: ‘DOUBLE_XMAX’ undeclared (first use in this function); did you mean ‘DBL_MAX’?
+   59 |   if (sum > DOUBLE_XMAX) {
+      |             ^~~~~~~~~~~
+      |             DBL_MAX
+make: *** [/usr/lib/R/etc/Makeconf:191: mean2.o] Error 1
+ERROR: compilation failed for package ‘matrixStats’
+```
+
+</details>
+
+The explanation for this error can be found in [the `matrixStats` release notes](https://github.com/HenrikBengtsson/matrixStats/blob/799669fc6de6f55a74cab06cc6a97634aa24ab0e/NEWS.md?plain=1#L111-L113), specifically the section for matrixStats 0.63.0:
+
+> * Updated native code to use the C99 constant `DBL_MAX` instead of legacy S
+constant `DOUBLE_XMAX`, which is planned to be unsupported in R (>= 4.2.0).
+
+## Some solutions
+
+### Larger binary store
```

Comment:
```suggestion
### Alternative package managers
```
Suggesting this as it is a more practical solution, based on saving more binary versions.